pkg/secrets/pushtofile/retrieve_secrets.go

```diff
@@ -0,0 +1,51 @@
+package pushtofile
+
+import (
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/access_token"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"strings"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/log/messages"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/secrets/clients/conjur"
+)
+
+var retrieveConjurSecretsFunc = conjur.RetrieveConjurSecrets
+
+func IterateOverGroups(accessToken access_token.AccessToken,
+	secretGroups *SecretGroups) error {
+
+	var err error
+	var retrieved map[string][]byte
+
+	for groupkey, group := range *secretGroups {
+		requiredSecrets := []string{}
+		for _, spec := range group.SecretSpecs {
+			requiredSecrets = append(requiredSecrets, spec.Id)
+		}
+		retrieved, err = RetrieveConjurSecretGroup(accessToken, requiredSecrets)
+
+		for _, spec := range group.SecretSpecs {
+			for id, secret := range retrieved {
+				if strings.Contains(id, spec.Id) {
+					(*secretGroups)[groupkey].SecretsMap[spec.Alias] = secret
+				}
+			}
+		}
+
+		if err != nil {
+			return log.RecordedError(messages.CSPFK052E, err.Error())
+		}
+
+	}
+	return err
+}
+
+func RetrieveConjurSecretGroup(
+	accessToken access_token.AccessToken,
+	variableIDs []string) (
+	map[string][]byte, error) {
+	accessTokenData, err := accessToken.Read()
+	if err != nil {
+		return nil, log.RecordedError(messages.CSPFK053E, err.Error())
```

Comment:
Can we simply reuse error `CSPFK003E` here, rather then creating a new error code?

Reply:
OK, will do. Maybe 2E fits better.